WordPress/Classes/ViewRelated/Comments/CommentViewController.m

```diff
@@ -658,23 +718,56 @@ - (void)dismissKeyboardIfNeeded:(id)sender
     [self.view endEditing:YES];
 }
 
+
+#pragma mark - Setters
+
+- (void)setComment:(Comment *)comment
+{
+    _comment = comment;
+    [self reloadData];
+}
+
+
 #pragma mark - Helpers
 
 - (BOOL)shouldAttachReplyTextView
 {
     // iPad: We've got a different UI!
-    return !([UIDevice isPad]);
+    return ![UIDevice isPad];
 }
 
 - (BOOL)shouldAttachSuggestionsTableView
 {
-    return ([self shouldAttachReplyTextView] && [[SuggestionService sharedInstance] shouldShowSuggestionsForSiteID:self.comment.blog.blogID]);
+    BOOL shouldShowSuggestions = [[SuggestionService sharedInstance] shouldShowSuggestionsForSiteID:self.comment.blog.blogID];
+    return self.shouldAttachReplyTextView && shouldShowSuggestions;
 }
 
-// if the post is not set for the comment, we don't want to show an empty cell for the post details
-- (BOOL)shouldShowHeaderForPostDetails
+- (void)reloadData
 {
-    return self.comment.post != nil;
+    // If we don't have the associated post, let's hide the Header
+    BOOL shouldShowHeader       = self.comment.post != nil;;
+
+    // Number of Rows
+    self.numberOfRows           = shouldShowHeader ? CommentsDetailsRowCount  : CommentsDetailsRowCount - 1;
+    self.rowNumberForHeader     = shouldShowHeader ? CommentsDetailsRowHeader : CommentsDetailsHiddenRowNumber;
```

Comment:
It seemed a bit weird to mix an enum and a const here.  I understand the intent after thinking about it.  Maybe add a comment for clarity?